core/src/main/scala/spire/math/Algebraic.scala

```diff
@@ -1,64 +1,165 @@
+  /**
+   * Returns an `Int` with the same sign as this algebraic number. Algebraic
+   * numbers support exact sign tests, so this is guaranteed to be accurate.
+   */
+  def signum: Int = expr.signum
 
-import algebraic._
+  /**
+   * Returns the sign of this Algebraic number. Algebraic numbers support exact
+   * sign tests, so this is guaranteed to be accurate.
+   */
+  def sign: Sign = Sign(signum)
 
+  /**
+   * Return a non-negative `Algebraic` with the same magnitude as this one.
+   */
+  def abs: Algebraic =
+    if (this.signum < 0) -this else this
 
-/**
- * An general Algebraic type. Can be used represent real numbers and approximate
- * them on-demand.
- */
-@SerialVersionUID(0L)
-final class Algebraic private (val expr: Expr[Algebraic])
-extends ScalaNumber with ScalaNumericConversions
-   with RealLike[Algebraic]
-   with BMFSSBound[Algebraic]
-   with BigDecimalApprox[Algebraic]
-   with FPFilter[Algebraic]
-   with ConstantFolder[Algebraic]
-   with BubbleUpDivs[Algebraic]
-   with PrettyToString[Algebraic]
-   with Ordered[Algebraic]
-   with Serializable {
-
-  val coexpr: Coexpr[Algebraic] = Algebraic.RealCoexpr
-
-  // ugh
-  override def byteValue(): Byte = fpf.toLong map (_.toByte) getOrElse super.toByte
-  override def shortValue(): Short = fpf.toLong map (_.toShort) getOrElse super.toShort
+  def unary_- : Algebraic =
+    new Algebraic(Expr.Neg(expr))
+
+  def +(that: Algebraic): Algebraic =
+    new Algebraic(Expr.Add(this.expr, that.expr))
+
+  def -(that: Algebraic): Algebraic =
+    new Algebraic(Expr.Sub(this.expr, that.expr))
+
+  def *(that: Algebraic): Algebraic =
+    new Algebraic(Expr.Mul(this.expr, that.expr))
+
+  def /(that: Algebraic): Algebraic =
+    new Algebraic(Expr.Div(this.expr, that.expr))
+
+  /**
+   * Returns an `Algebraic` whose value is just the integer part of
+   * `this / that`. This operation is exact.
+   */
+  def quot(that: Algebraic): Algebraic =
+    this /~ that
+
+  /** An alias for [[quot]]. */
+  def /~(that: Algebraic): Algebraic =
+    Algebraic((this / that).toBigInt)
+
+  /**
+   * Returns an `Algebraic` whose value is the difference between `this` and
+   * `(this /~ that) * that` -- the modulus.
+   */
+  def mod(that: Algebraic): Algebraic =
+    this % that
+
+  /** An alias for [[mod]]. */
+  def %(that: Algebraic): Algebraic =
+    this - (this /~ that) * that
+
+  /** Returns the square root of this number. */
+  def sqrt: Algebraic = nroot(2)
+
+  /** Returns the cube root of this number. */
+  def cbrt: Algebraic = nroot(3)
+
+  /** Returns the `k`-th root of this number. */
+  def nroot(k: Int): Algebraic = if (k < 0) {
+    new Algebraic(Expr.Div(Expr.ConstantLong(1), Expr.KRoot(this.expr, -k)))
+  } else if (k > 0) {
+    new Algebraic(Expr.KRoot(this.expr, k))
+  } else {
+    throw new ArithmeticException("divide by zero (0-root)")
+  }
+
+  /** Raise this number to the `k`-th power. */
+  def pow(k: Int): Algebraic =
```

Comment:
Would be nice to test this with large values (performance permitting) and also some of the exceptional cases (0, Int.MinValue, etc).